### PR TITLE
make the motion sensor dynamic.

### PR DESCRIPTION
**Pull request by jgstroud**
Only enable the service if a motion sensor is detected. By default the service will be disabled.  On the first motion detect message, set a flag to enable the service and reboot to trigger the addition of the new service in HK.  Will be permanently enabled after.

### CODE REVIEW

**Comment by jgstroud** [comment]
Note, this change will almost certainly require you to repair after loading this for the first time.

**Comment by jgstroud** [comment]
Fixes https://github.com/ratgdo/homekit-ratgdo/issues/64

**Comment by jgstroud** [comment]
After you load this firmware, your device will show up with no motion sensor.  If you have a motion sensor, then after pairing your device with HK, trigger the motion sensor.  A short time later (it can take a minute), the motion device will show up in HK.

**Comment by dkerr64** [comment]
Given that the Arduino HomeKit stack is buggy (It regularly and randomly un-pairs for me), I think it wise to hold off merging this until we have the Espressif HomeKit stack... which will hopefully be more reliable.

My concern is the need to go and activate the motion detector every re-pair (which for me is too often).

Also, Apple Home deletes any automations attached to an accessory if that accessory is removed.  So we want this to be a once-and-done thing.

Or maybe I'm missing something... once a motion detector is found, is that saved for good in EEPROM (like WiFi SSID/PW) or filesystem such that un-pair and re-pair will not require a new discovery of the motion detector?

**Comment by donavanbecker** [comment]
@dkerr64 Why not merge and release updates until that point?

**Review by donavanbecker — APPROVED** [review]

**Comment by dkerr64** [comment]
> @dkerr64 Why not merge and release updates until that point?

I'm not against merging updates.  I'm concerned that I will have to manually trigger motion to re-discover the motion detector every time I have to re-pair.  If it is a once-and-done then I'm happy.

**Comment by jgstroud** [comment]
It's saved in flash.  So it's a one time activation.  Don't have to reenabled after repairing.

**Review by donavanbecker — APPROVED** [review]

File: src/utilities.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 Brandon Matthews <thenewwazoo@optimaltour.us>
+// All rights reserved. GPLv3 License
+
+#include "utilities.h"
+#include "log.h"
+#include "LittleFS.h"
+
+
+void sync_and_restart() {
```
Comment (thenewwazoo) [inline src/utilities.cpp:9]: This function calls the GDO Security+ 2.0 "sync" operation to mind. maybe "graceful_restart"?

**Pull request merged**